libs/pubsub/pubsub.go

```diff
@@ -368,6 +368,7 @@ func (s *Server) run() {
 	ctx, cancel := context.WithCancel(context.Background())
 	s.done = ctx.Done()
 	s.stop = cancel
+	queue := s.queue
```

Comment:
looks like we could just use the context passed to OnStart for this, which might help?

Reply:
It's unrelated, but yes, we can certainly do that here as well. I'll add that if, as I suspect, this resolves @cmwaters's issue.